src/main/java/duke/utility/Storage.java

```diff
@@ -0,0 +1,128 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * <h2>Storage</h2>
+ * A class that helps read previous task logs from local memory and write to that task log based on user input.
+ */
+
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Creates a link to a locally saved task log or creates a new file for future logging of tasks.
+     * @param filePath the path to the file to read from and write to or location to create the new file if it does not
+     *                exist.
+     * @throws IOException if file does not exist and cannot be created.
+     */
+    public Storage(String filePath) throws IOException {
+        this.filePath = filePath;
+        // try to load the file, if not, create it
+        File previousLog = new File(this.filePath);
+        if (!previousLog.createNewFile()) { // file already exists
+            System.out.println("Successfully established connection with file " + filePath
+                    + "\nPrevious duke.task log imported."
+                    + "\nAll changes to task log will be saved there.");
+        } else { // new file is created
+            System.out.println("Specified file not found.\nNew file created: " + filePath
+                    + "\nTask log will be saved there.");
+        }
+    }
+
+    /**
+     * Parses the locally saved task log from the file path provided and generates a list of tasks to import.
+     * @return a list of tasks generated from parsing the locally saved task log.
+     * @throws FileNotFoundException if the stipulated file can neither be read from (e.g. does not exist) nor
+     * be created.
+     */
+    public List<Task> loadPreviousTasks() throws FileNotFoundException {
+        List<Task> prevTasks = new ArrayList<Task>();
+        Scanner sc = new Scanner(new File(this.filePath));
+        while (sc.hasNextLine()) {
+            String task = sc.nextLine();
+            String[] tokens = task.split(";");
+            System.out.println(Arrays.toString(tokens));
+            boolean isCompleted = tokens[2].equals("T");
+            String taskName = tokens[1];
+            // format: duke.task type | isCompleted | event name | date/time
+            switch (tokens[0]) {
+            case "T":
+                prevTasks.add(ToDo.createTask(taskName, isCompleted));
+                break;
+            case "E":
+                prevTasks.add(Event.createTask(taskName, isCompleted, LocalDateTime.parse(tokens[3].trim(),
+                        DateTimeFormatter.ISO_LOCAL_DATE_TIME)));
+                break;
+            default: // "D"
+                prevTasks.add(Deadline.createTask(taskName, isCompleted, LocalDateTime.parse(tokens[3].trim(),
+                        DateTimeFormatter.ISO_LOCAL_DATE_TIME)));
+                break;
+            };
+        }
+        return prevTasks;
+    }
+
+    void append(String e, String f, String taskName, String dateTime) throws IOException {
```

Comment:
perhaps add some javadocs for the methods below like append and changeTaskLogToCompleted